src/doc/src/reference/workspaces.md

```diff
@@ -82,7 +82,25 @@ default-members = ["path/to/member2", "path/to/member3/foo"]
 
 When specified, `default-members` must expand to a subset of `members`.
 
+### The `workspace.metadata` table
+
+Like the [`package.metadata`][package-metadata] table, the `workspace.metadata`
+table is ignored by Cargo and will not be warned about. This section can be
+used for tools that would like to store workspace configuration in
+`Cargo.toml`. For example:
+
+```toml
+[workspace]
+members = ["member1", "member2"]
+
+[workspace.metadata.webcontents]
+root = "path/to/webproject"
+tool = ["npm", "run", "build"]
+# ...
+```
+
```

Comment:
I would suggest adding some general guidance here, suggesting that while the format of the `package.metadata` and `workspace.metadata` sections are not specified, tools may wish to use them in a consistent fashion, such as taking data from the workspace if not present in the package, if that makes sense for the tool in question.

Reply:
I was wondering to what degree we should make suggestions. I would imagine something like having the top-level metadata table be indexed by a tool-specific unique string, and to put other data below that, but did not know if that would just be adding documentation noise.

Since this incorporates information from two different sources ([package] and [workspace]), it seems like that kind of suggestion would also be best put either in both places, or a common place that both reference. Would that be a good reason for adding a new section? Is there an existing file which would be good to give some suggestions to the user?

Reply:
I would suggest having a new section for `workspace.metadata` in the context of `workspace`, cross-linking between them, and having a one-sentence suggestion for consistency associated with both cross-links.

Reply:
Added. I hope you don't mind me using your words for this.